filebeat/beater/filebeat.go

```diff
@@ -235,8 +237,11 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 		return err
 	}
 
+	fb.pipeline = pipetool.WithPipelineEventCounter(b.Publisher, wgEvents)
+	fb.pipeline = pipetool.WithDefaultGuarantees(fb.pipeline, beat.GuaranteedSend)
```

Comment:
When doing this change I noticed wgEvents was propably not hooked up correctly. This PR might fix an issue or flaky tests with shutdown timeout not working correctly.